torchmetrics/audio/sdr.py

```diff
@@ -0,0 +1,138 @@
+# Copyright The PyTorch Lightning team.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+from typing import Any, Callable, Optional
+
+from torch import Tensor, tensor
+
+from torchmetrics.functional.audio.sdr import sdr
+from torchmetrics.metric import Metric
+
+
+class SDR(Metric):
+    r"""Signal to Distortion Ratio (SDR) [1,2,3]
+
+    .. note:: 1) using this metrics requires you to have ``fast-bss-eval`` install. Either install as ``pip install
+        torchmetrics[audio]`` or ``pip install fast-bss-eval``
+        2) preds and target need to have the same dtype, otherwise target will be converted to preds' dtype
+        3) when pytorch<1.8.0, numpy will be used to calculate this metric, which causes ``sdr`` non-differentiable
+
+    Forward accepts
+
+    - ``preds``: shape ``[..., time]``
+    - ``target``: shape ``[..., time]``
+
+    Args:
+        use_cg_iter:
+            If provided, an iterative method is used to solve for the distortion \
+            filter coefficients instead of direct Gaussian elimination. \
+            This can speed up the computation of the metrics in case the filters \
+            are long. Using a value of 10 here has been shown to provide \
+            good accuracy in most cases and is sufficient when using this \
+            loss to train neural separation networks.
```

Comment:
```suggestion
            If provided, an iterative method is used to solve for the distortion
            filter coefficients instead of direct Gaussian elimination.
            This can speed up the computation of the metrics in case the filters
            are long. Using a value of 10 here has been shown to provide
            good accuracy in most cases and is sufficient when using this
            loss to train neural separation networks.
```
pls remove all these even from the following docstrings

Reply:
removed